cassandra/src/main/scala/zio/flow/internal/CassandraKeyValueStore.scala

```diff
@@ -0,0 +1,219 @@
+package zio.flow.internal
+
+import com.datastax.oss.driver.api.core.`type`.DataTypes
+import com.datastax.oss.driver.api.core.cql.{AsyncResultSet, Row, SimpleStatement}
+import com.datastax.oss.driver.api.core.servererrors.InvalidQueryException
+import com.datastax.oss.driver.api.core.{CqlIdentifier, CqlSession}
+import com.datastax.oss.driver.api.querybuilder.QueryBuilder.literal
+import com.datastax.oss.driver.api.querybuilder.{QueryBuilder, SchemaBuilder}
+
+import java.io.IOException
+import java.nio.ByteBuffer
+import zio.{Chunk, Has, IO, URLayer, ZIO, ZRef}
+import zio.stream.ZStream
+
+final class CassandraKeyValueStore(session: CqlSession) extends KeyValueStore {
+  import CassandraKeyValueStore._
+
+  private val keyspace =
+    session.getKeyspace
+      .orElse(null: CqlIdentifier)
+
+  override def put(
+    namespace: String,
+    key: Chunk[Byte],
+    value: Chunk[Byte]
+  ): IO[IOException, Boolean] = {
+    val insert =
+      QueryBuilder
+        .insertInto(keyspace, tableOf(namespace))
```

Comment:
We may want to have one table with a column for the namespace instead of separate tables for each namespace. We should expect to have a large number of namespaces for a ZIO Flow application. For example each `DurablePromise` we create has its own namespace.

Reply:
@adamgfraser Do you mean something like this?
```cql
CREATE TABLE _zflow_key_value_store (
  namespace        VARCHAR,
  kv_store_key     BLOB,
  kv_store_value  BLOB,
  PRIMARY KEY ((namespace, kv_store_key))
)
```

Essentially `(namespace, kv_store_key)` is the composite partition key of `_zflow_key_value_store` table.

Reply:
@ithinkicancode Yes exactly.

Reply:
New implementation is using this data model.